zerver/openapi/zulip.yaml

```diff
@@ -12125,6 +12202,54 @@ paths:
                 oneOf:
                   - $ref: "#/components/schemas/InvalidPushDeviceTokenError"
                   - $ref: "#/components/schemas/InvalidRemotePushDeviceTokenError"
+  /mobile_push/register:
+    post:
+      operationId: register-push-account
+      summary: Register push account for E2EE notifications
+      tags: ["mobile"]
+      description: |
+        Register an account on an install of the app to receive
+        end-to-end encrypted mobile push notifications.
+
+        **Changes**: New in Zulip 11.0 (feature level ZF-1653f2).
+      requestBody:
+        required: true
+        content:
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              properties:
+                token_kind:
+                  description: |
+                    Whether the token was generated by FCM or APNs.
+
+                    - APNs = 1
+                    - FCM = 2
+                  type: integer
+                  example: 1
+                push_account_id:
+                  description: |
+                    The ID of the account on an install of the app to register.
+                  type: integer
+                  example: 9006
+                push_public_key:
+                  description: |
+                    The public key to use to encrypt the notifications.
+                  type: string
+                  example: "push-public-key"
+                bouncer_public_key:
+                  description: |
+                    The public key used to encrypt `encrypted_push_registration`.
+                  type: string
+                  example: "bouncer-public-key"
+                encrypted_push_registration:
+                  description: |
+                    The encrypted data to be used by bouncer.
```

Comment:
Similarly, this should include the algorithm for generating this (libnacl, b64 encoding, etc.). You should be able to copy this from the spec.